test/suites/network/offline.robot

```diff
@@ -0,0 +1,212 @@
+        Log    Default MicroShift config not found, ignoring
+    END
+    Set Suite Variable    ${DEFAULT_RESOLV_FILE}    ${data}
+
+    Configure Guest Lo Connection
+    Write To File    ${GUEST_NAME}    /etc/hosts    \n10.44.0.1 ${GUEST_NAME}\n    append=${TRUE}
+    Write To File    ${GUEST_NAME}    /etc/resolv.conf    \nnameserver 10.44.1.1\n    append=${TRUE}
+
+Configure Guest Lo Connection
+    [Documentation]    Configure the guest's loopback connection
+    Run Guest Process    ${GUEST_NAME}    nmcli    connection    modify    lo    +ipv4.addresses    10.44.0.1/32
+    Run Guest Process    ${GUEST_NAME}    nmcli    connection    modify    lo    autoconnect    yes
+    Run Guest Process    ${GUEST_NAME}    hostnamectl    set-hostname    ${GUEST_NAME}
+
+Restore Guest Environment
+    [Documentation]    Restore the guest's configuration files
+    Write To File    ${GUEST_NAME}    /etc/hosts    ${DEFAULT_HOSTS_FILE}
+
+    ${len}=    Get Length    ${DEFAULT_RESOLV_FILE}
+    IF    ${len} > 0
+        Write To File    ${GUEST_NAME}    /etc/resolv.conf    ${DEFAULT_RESOLV_FILE}
+    ELSE
+        Run Guest Process    ${GUEST_NAME}    rm    /etc/resolv.conf
+    END
+
+    Run Guest Process    ${GUEST_NAME}    nmcli    connection    modify    lo    -ipv4.addresses    10.44.0.1/32
+    Run Guest Process    ${GUEST_NAME}    nmcli    connection    modify    lo    connection.autoconnect    no
+
+Save Default MicroShift Config
+    [Documentation]    Save the default MicroShift config for offline start.    MicroShift fails on first boot when
+    ...    offline, so ignore the file not exist error.
+    ${data}=    Set Variable    ${EMPTY}
+    TRY
+        ${data}=    Run Keyword
+        ...    Read From File    ${GUEST_NAME}    /etc/microshift/config.yaml
+    EXCEPT    *No such file or directory*    type=glob
+        Log    Default MicroShift config not found, ignoring
+    END
+    Set Suite Variable    ${DEFAULT_MICROSHIFT_CONFIG}    ${data}
+
+Restore Default MicroShift Config
+    [Documentation]    Restore the default MicroShift config for offline start.
+    ${len}=    Get Length    ${DEFAULT_MICROSHIFT_CONFIG}
+    IF    ${len} > 0
+        Write To File    ${GUEST_NAME}    /etc/microshift/config.yaml    ${DEFAULT_MICROSHIFT_CONFIG}
+    ELSE
+        ${result}    ${ignore}=    Run Guest Process    ${GUEST_NAME}    rm    /etc/microshift/config.yaml
+        Log Many    ${result["stdout"]}    ${result["stderr"]}    ${result["rc"]}
+        Should Be Equal As Integers    ${result["rc"]}    0
+    END
+
+Restart MicroShift
+    [Documentation]    Restart the MicroShift systemd service
+    ${result}    ${ignore}=    Wait Until Keyword Succeeds    5x    5s
+    ...    Run Guest Process    ${GUEST_NAME}    systemctl    restart    microshift.service
+    Log Many    ${result["stdout"]}    ${result["stderr"]}    ${result["rc"]}
+    Should Be Equal As Integers    ${result["rc"]}    0
+
+Restart Greenboot Healthcheck
```

Comment:
Is this better named as `Restart Greenboot And Wait Health Check To Exit`?
Otherwise, I had to look in the implementation to understand if there's a wait for success.